compiler/testData/diagnostics/tests/controlFlowAnalysis/reassignmentInTryCatchWithJumps.fir.kt

```diff
@@ -23,14 +23,14 @@ fun f2(flag: Boolean) {
         val n: Int
         try {
             if (flag) {
-                n = 1
+                <!VAL_REASSIGNMENT!>n<!> = 1
```

Comment:
Due to the `break` below, this one wouldn't be revisited via the outer `while` loop. I'll keep this in mind while re-working https://github.com/JetBrains/kotlin/pull/3773

Reply:
We shouldn't have reassignment here

Reply:
But *should* have it reported in catch, both in `f1` and in `f2`

Reply:
> We shouldn't have reassignment here

Yep, no longer report anything here.

> But _should_ have it reported in catch, both in f1 and in f2

Nothing reported at all, but I'm not sure why we _should_ report `val` reassignment in this test file:
https://github.com/JetBrains/kotlin/blob/master/compiler/testData/diagnostics/tests/controlFlowAnalysis/reassignmentInTryCatchWithJumps.kt

So, every `n = 3` in `catch` clause can be reached if, e.g., there is OOM thrown before `n = 1`.

Reply:
Okay, after thinking about these a bit more, I have to admit we have no false positives (which is good), but yet not reporting those assignments in `catch` clauses are false negatives because every `n = 3` in `catch` clause can be reached if, e.g., there is OOM thrown _after_ `n = 1`.

The pattern of test functions in this file is that `try` main block has a `when` expression whose main part will escape the `try` main block (through `return`, `continue`, or `break`). Because of that, the `WhenBranchResultExitNode` is marked as dead, and the edge from it is also `DeadForward`, which gets that node discarded when collecting previous CFG nodes for the `when` exit node. Thus, assignment info inside `if` is always discarded. Therefore, assignment info inside `catch` clause is always `AT_MOST_ONCE`, which should be `AT_LEAST_ONE` instead.

So, this is related TODOs I left at https://github.com/JetBrains/kotlin/commit/440cf7888460aa4d691c15c3dfe358465a1e59fb: when adding jump nodes, if they're inside try-catch-finally, we should be more careful. As shown in this case, we need an edge towards the next node (for unexpected error at any place) in addition to the edge toward the jump target (we're adding this only for now). I'll keep this thread of comments open, not to forget the issue, and will try a bit more while waiting for the reviews. But, since there are no false errors, this PR should be good to go too.